webclient/core/import_data_mappers.py

```diff
@@ -0,0 +1,1522 @@
+                start = datetime.datetime.strptime(start, "%Y-%m-%d").date()
+                end = datetime.datetime.strptime(end, "%Y-%m-%d").date()
+                return DateRange(start, end)
+        raise ImportDataError(f"{_('core_admin.DateRangeImportField.message.invalid_date_range_value')}: {value}")
+
+
+class LookupImportField(BaseImportField):
+    """
+    Class for import fields that should contain a reference to another model instance.
+    """
+
+    records = []
+
+    def __init__(self, lookup_model_classes=None, lookup_field_name: str = "ident_cely"):
+        super().__init__()
+        if not isinstance(lookup_model_classes, Iterable):
+            self.lookup_model_class_list = [
+                lookup_model_classes,
+            ]
+        elif lookup_model_classes is None:
+            self.lookup_model_class_list = []
+        else:
+            self.lookup_model_class_list = lookup_model_classes
+        self.lookup_field_name = lookup_field_name
+        self._instance_value = None
+
+    @property
+    def instance_value(self):
+        return self._instance_value
+
+    def _process_value(self, value):
+        """
+        Processes the value by checking if it exists in the database or in the imported records. If yes, it returns
+        the record. If the referenced record does not exist, it raises ImportDataMissingReferencedValueError.
+        """
+
+        if str(value).lower() == "nan" or value is None or len(str(value)) == 0:
+            return None
+        for current_class in self.lookup_model_class_list:
+            saved_records_query = current_class.objects.filter(**{self.lookup_field_name: value})
+            if saved_records_query.exists():
+                self._instance_value = saved_records_query.first()
+                return value
+            filtered_records = [
+                record
+                for record in self.records
+                if isinstance(record, current_class) and getattr(record, self.lookup_field_name, None) == value
+            ]
+            if len(filtered_records) == 1:
+                self._instance_value = filtered_records[0]
+                return value
+        raise ImportDataMissingReferencedValueError(
+            value, ", ".join([current_class.__name__ for current_class in self.lookup_model_class_list])
+        )
+
+
+class RuianLookupImportField(LookupImportField):
+    """
+    Based on the LookupImportField, this class is used for importing data from RUIAN data. It strips
+    the "ruian-" prefix from the value and converts it to an integer.
+    """
+
+    @LookupImportField.value.setter
+    def value(self, value):
+        if isinstance(value, str):
+            match = re.match(r"ruian-(\d+)", value)
+            value = int(match.group(1))
+        LookupImportField.value.fset(self, value)
+
+
+class KomponentaLookupImportField(LookupImportField):
+    """
+    Class for import field with referenced models for Komponenta.
+    """
```

Comment:
Nepoužívá se jen pro komponenty - mělo by se lépe pojmenovat a popsat. Také bych byl asi explicitnější pro dílčí varianty použití (komponenta, soubor, historie), ať je to robustnější a lépe čitelné.